templates/build/template.yml

```diff
@@ -0,0 +1,172 @@
+  description: Git source URI for HTTPD application
+  displayName: HTTPD application source url
+  value: https://github.com/logandonley/infographic.git
+  required: true
+- name: HTTPD_APPLICATION_SOURCE_REPOSITORY_REF
+  description: HTTPD application source branch/tag reference.
+  displayName: HTTPD application source branch/tag reference.
+  value: master
+  required: false
+- name: HTTPD_APPLICATION_CONTEXT_DIR
+  description: Path within Git project to build containing the HTTPD application;
+    empty for root project directory.
+  displayName: HTTPD application context directory.
+  value: website
+  required: true
+- name: GITHUB_WEBHOOK_SECRET
+  displayName: GitHub Webhook Secret
+  description: A secret string used to configure the GitHub webhook.
+  generate: expression
+  from: "[a-zA-Z0-9]{40}"
+objects:
+- kind: ImageStream
+  apiVersion: v1
+  metadata:
+    name: sclorg-httpd-container
+    namespace: "${NAMESPACE}"
+    labels:
+      build: "${HTTPD_BUILDER_NAME}"
+  spec:
+    dockerImageRepository: registry.access.redhat.com/rhscl/httpd-24-rhel7
+- kind: ImageStream
+  apiVersion: v1
+  metadata:
+    name: "${HTTPD_BUILDER_NAME}"
+    namespace: "${NAMESPACE}"
+    labels:
+      build: "${HTTPD_BUILDER_NAME}"
+- kind: ImageStream
+  apiVersion: v1
+  metadata:
+    name: "${HTTPD_APPLICATION_NAME}"
+    namespace: "${NAMESPACE}"
+    labels:
+      application: "${HTTPD_APPLICATION_NAME}"
+- kind: BuildConfig
+  apiVersion: v1
+  metadata:
+    labels:
+      build: "${HTTPD_APPLICATION_NAME}"
+      name: "${HTTPD_APPLICATION_NAME}-pipeline"
+    name: "${HTTPD_APPLICATION_NAME}-pipeline"
+    namespace: "${NAMESPACE}"
+  spec:
+    nodeSelector: {}
+    output: {}
+    postCommit: {}
+    resources: {}
+    runPolicy: Serial
+    source:
+      contextDir: "${HTTPD_APPLICATION_CONTEXT_DIR}"
+      git:
+        ref: "${HTTPD_APPLICATION_SOURCE_REPOSITORY_REF}"
+        uri: "${HTTPD_APPLICATION_SOURCE_REPOSITORY_URL}"
+      type: Git
+    strategy:
+      jenkinsPipelineStrategy:
+        jenkinsfilePath: Jenkinsfile
+      type: JenkinsPipeline
+    triggers:
+    - github:
+        secret: "${PIPELINE_GITHUB_WEBHOOK_SECRET}"
+      type: GitHub
+    - type: ConfigChange
+- kind: BuildConfig
+  apiVersion: v1
+  metadata:
+    name: "${HTTPD_BUILDER_NAME}"
+    namespace: "${NAMESPACE}"
+    labels:
+      build: "${HTTPD_BUILDER_NAME}"
+  spec:
+    triggers:
+    - type: ConfigChange
+    - type: ImageChange
+      imageChange: {}
+    source:
+      type: Git
+      git:
+        uri: "${HTTPD_BUILDER_SOURCE_REPOSITORY_URL}"
+        ref: "${HTTPD_BUILDER_SOURCE_REPOSITORY_REF}"
+      contextDir: "${HTTPD_BUILDER_CONTEXT_DIR}"
+    strategy:
+      type: Docker
+      dockerStrategy:
+        from:
+          kind: ImageStreamTag
+          name: sclorg-httpd-container:2.4
+    output:
+      to:
+        kind: ImageStreamTag
+        name: "${HTTPD_BUILDER_NAME}:latest"
```

Comment:
Can we please add an `IMAGE_TAG` parameter instead of `latest` and default the parameter to `latest`? 
Similar to what we did here:
https://github.com/redhat-cop/cluster-lifecycle/blob/master/deploy/cakephp/cakephp-template.yml#L111